src/components/BuildMetrics/BuildMetrics.tsx

```diff
@@ -166,91 +164,123 @@ const navigationOptions = [
   { id: 10, name: '10th' },
 ];
 
-const getNavigationIdByName = (name: string) => navigationOptions.find((option) => option.name === name)?.id!;
-
 /**
- * Load Build Metrics for specified builds.
+ * Get navigation id by navigation option name
  *
- * @param {Object[]} builds - List of Builds.
+ * @param name - Navigation option name
  */
-const transferBuildsToBuildId = (builds?: Array<Build>) => {
-  if (builds) {
-    return builds.map((build) => {
-      return build.id.toString();
-    });
-  }
+const getNavigationIdByName = (name: string): number => {
+  const foundOption = navigationOptions.find((option) => option.name === name);
+  return foundOption ? foundOption.id : 1;
 };
 
-const MetricsPopoverContent = (metricsTooltipList: Array<IMetricsTooltip>) => {
-  return (
-    <div>
-      Each metric consists of several subtasks:
-      {metricsTooltipList.map((metricsTooltip) => {
-        return (
-          <dl className={styles['pnc-popover-paragraph']} key={metricsTooltip.label}>
-            <dt>
-              <b>{metricsTooltip.label}</b>
-            </dt>
-            <dd>{metricsTooltip.description}</dd>
-          </dl>
-        );
-      })}
-    </div>
-  );
+/**
+ * Convert builds into an array of build ids as strings
+ *
+ * @param builds - List of builds
+ */
+const getBuildIds = (builds?: Build[]): string[] | undefined => builds?.map((build) => build.id.toString());
+
+/**
+ * Render the popover content for metrics tooltip
+ *
+ * @param metricsTooltipList - List of metric tooltips
+ */
+const MetricsPopoverContent = (metricsTooltipList: IMetricsTooltip[]) => (
+  <div>
+    Each metric consists of several subtasks:
+    {metricsTooltipList.map((tooltip) => (
+      <dl className={styles['pnc-popover-paragraph']} key={tooltip.label}>
+        <dt>
+          <b>{tooltip.label}</b>
+        </dt>
+        <dd>{tooltip.description}</dd>
+      </dl>
+    ))}
+  </div>
+);
+
+/**
+ * Filter (sample) builds from the provided array
+ *
+ * @param buildsArray - Array of builds
+ * @param nth - Sample every nth build
+ * @param max - Maximum number of builds to display
+ */
+const sampleBuilds = (buildsArray: Build[], nth: number, max: number = BUILDS_DISPLAY_LIMIT): Build[] => {
+  const sampled: Build[] = [];
+  for (let i = 0; i < buildsArray.length; i += nth) {
+    sampled.push(buildsArray[i]);
+  }
+  return sampled.slice(0, max);
```

Comment:
instead of slice, you can extend the `i < buildsArray.length` condition

Reply:
Hi @patrikk0123 , this has been updated, do you have any other reviews over this PR? Thanks!

Reply:
I approve